gen/src/rfcgen.act

```diff
@@ -898,12 +916,36 @@ def gen(stmts: dict[str, Stmt]) -> list[str]:
         stmt = stmts[stmt_name]
         arg_name = stmt.argument_name
         have_children = False
-        cattrs = []
+
+        # Sort the list of attributes so that attributes with cardinality 1 are
+        # first, plus a custom order for "module" substatements so that the
+        # module layout conforms to the "typical layout" outlined in the RFC.
+        module_order = [
+            "yang-version", "namespace", "prefix", "import", "include",
+            "organization", "contact", "description", "reference", "revision"
+        ]
+
+        def _compute_priority(stmt_name: str, attr: str, card: str, parent_arg_name: ?str) -> int:
+            if attr == parent_arg_name:
+                # Argument to parent statement is always first
+                return -1
+            base = 0 if card == "1" else 1
+            if (stmt_name == "module" or stmt_name == "submodule") and attr in module_order:
+                # Insert the index as well, shifting bigger cardinality first
+                return base * 100 + module_order.index(attr)
+            return base * 1000
+
+        cattrs2 = []
```

Comment:
I don't think cattrs2 should come before cattrs2. If we are to do the prime style
```haskell
let a   = ...
    a'  = a + bopp
    a'' = f(a')
```
then they need to come in order, i.e.
- cattrs
- cattrs2
- cattrs3

Otherwise I prefer like
- unsorted_cattrs
- cattrs = sorted(unsorted_cattrs)

or something like that